README


This archive contains the distribution Markdent,
version 0.22:

  An event-based Markdown parser toolkit

This software is copyright (c) 2012 by Dave Rolsky.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.




Changes
0.25   2015-01-04

- HTML output now uses the HTML5 doctype. In addition, table cell alignment is
  set using a style tag in generated HTML, rather than the deprecated-in-HTML5
  "align" attribute.

- The Markdent::Handler::HTMLStream::Document and Markdent::Simple::Document
  classes now accept optional charset and language parameters.

- The HTML output for table cells was broken. Cells always ended with </th>
  even for body cells. Reported by Polina Shubina. GitHub PR #2.

- Replaced Class::Load with Module::Runtime.


0.24   2014-01-25

- Shut up enum warnings from the latest Moose (2.1200). Patch by Zoffix. RT
  #92274.


0.23   2013-12-13

- Added support for drunken (42, 5, 3) and lazy (1, 1, 1) ordered lists, per
  the Markdown spec -
  http://daringfireball.net/projects/markdown/syntax#list. Based on tests and
  patch from Jason McIntosh.


0.22   2012-07-22

* All dialects except the Standard dialect are now roles. This makes it
  possible to create a parser that parses multiple dialects. It's possible for
  dialects to have conflicting parsing rules, in which case the results are
  unpredictable.

  This is a backwards incompatible change, but should not break most code,
  unless you have implemented your own dialect, or you were explicitly setting
  a block or span parser class to something like

      block_parser_class => 'Markdent::Dialect::Theory::BlockParser'

  But there was really no good reason to do this.

- The Standard dialect block/span parser classes are now named
  Markdent::Parser::BlockParser and ::SpanParser.

- The use of a "dialect" as a named parameter has been replaced by
  "dialects". The "dialects" parameter can be either a string or an array ref
  of strings. However, the "dialect" parameter is still accepted for backwards
  compatibility. This may be removed in the future.

- The Standard dialect now parses two spaces at the end of a line as a line
  break. There is a new Markdent::Event::LineBreak class to go with this.

- Improve the bin/markdent-html script. It now has a --help option. You can
  pass it Markdown text via the CLI --text option, rather than only via a
  --file.

- Implemented most of GitHub-flavored Markdown as a dialect.


0.21   2012-06-22

- The StartListItem event now has a bullet attribute. This contains the text
  of the bullet, something like "*" or "1.". Requested by Steven Haryanto. RT
  #77893.


0.20   2012-05-18

- Work around breakage introduced with MooseX::Types 0.32.


0.19   2012-03-04

- Some modules were missing a version number in the last release.


0.18   2012-03-04

- The Markdent::Handler::HTMLStream::Document and ::Fragment modules did not
  detect when streaming to a file handle failed. This is because HTML::Stream
  does not check that it's calls to ->print succeed. This has been fixed by
  wrapping file handles (native or IO::Handle) in an object which checks that
  print succeeds.

- Distro/packaging cleanup.


0.17   2010-09-27

- Avoid warnings from the next Moose.

- Added a Null handler.


0.16   2010-06-03

- The last release (and others?) did not include a $VERSION in its
  modules. Doh.


0.15   2010-06-03

- Even more table parsing improvements. Now the parse speed is approximately
  50 times faster than 0.13.


0.14   2010-06-03

- Greatly improved table parsing speed. Large tables took a very long time to
  parse, as the parser ended up backtracking way too much. Thanks to Yuval
  Kogman for his suggestions. Now large tables parse approximately 10 times
  faster.


0.13   2010-06-01

- Fixed handling of a very pathological case for table parsing, which caused
  the parser to die. Instead, it will now keep going. Your output may be
  bizarre, but the parser should not die.

- Fixed handling of a tab character next to a cell delimiter. This caused the
  parser to become very confused and go into an endless loop. Now such tabs
  are simply stripped from the output (just like other whitespace next to a
  cell delimiter).


0.12   2010-05-25

- Allow setting the handler class when calling Test::Markdent::parse_ok().


0.11   2010-05-25

- Moved Test::Markdent into lib (from t/lib), since it is useful to anyone who
  wants to write a Markdent dialect.


0.10   2010-05-19

- Split all HTML generating code into ::Document and ::Fragment classes. This
  means that Markdent::Handler::HTMLStream is now
  Markdent::Handler::HTMLStream::Document and
  Markdent::Handler::HTMLStream::Fragment. Similarly, Markdent::Simple is now
  Markdent::Simple::Document and Markdent::Simple::Fragment.

  This was done to make it easy to turn a snippet of Markdown into a snippet
  of HTML. Previously, you could only generate a complete HTML document.


0.09   2010-02-15

- The Theory dialect parser would parse text in brackets as a table caption
  even if no table followed, for example:

  [Not a caption]

  Just some text


0.08   2009-12-06

- Added a missing dependency, MooseX::Role::Parameterized.


0.07   2009-11-27

- You can now include a header marker at the beginning and/or end of the
  table. This matches how MySQL outputs tables from its CLI tool.

  +------+-------------+------------------------------+--------+
  | id   | name        | description                  | price  |
  +------+-------------+------------------------------+--------+
  |    1 | gizmo       | Takes care of the doohickies |   1.99 |
  |    2 | doodad      | Collects *gizmos*            |  23.80 |
  |   10 | dojigger    | Foo                          | 102.98 |
  | 1024 | thingamabob | Self-explanatory, no?        |   0.99 |
  +------+-------------+------------------------------+--------+

  Suggested by David Wheeler.


0.06   2009-11-27

- Added a new handler, Markdent::Handler::HTMLFilter, which removes all HTML
  events (except for entities) from the event stream.

- Added a new role, Markdent::Role::FilterHandler, to make it easier to write
  filtering handlers.


0.05   2009-11-27


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.012.
use strict;
use warnings;



use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "An event-based Markdown parser toolkit",
  "AUTHOR" => "Dave Rolsky <autarch\@urth.org>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Markdent",
  "EXE_FILES" => [
    "bin/markdent-html"
  ],
  "LICENSE" => "perl",
  "NAME" => "Markdent",
  "PREREQ_PM" => {
    "Class::Load" => 0,
    "Digest::SHA1" => 0,
    "Exporter" => 0,
    "File::Slurp" => 0,
    "Getopt::Long::Descriptive" => 0,
    "HTML::Stream" => 0,
    "IO::Handle" => 0,
    "List::AllUtils" => 0,
    "Moose" => "0.92",
    "Moose::Meta::Class" => 0,
    "Moose::Role" => 0,
    "MooseX::Getopt::Dashes" => 0,
    "MooseX::Params::Validate" => 0,
    "MooseX::Role::Parameterized" => 0,
    "MooseX::SemiAffordanceAccessor" => "0.05",
    "MooseX::StrictConstructor" => "0.08",
    "MooseX::Types" => "0.20",
    "MooseX::Types::Combine" => 0,
    "MooseX::Types::Moose" => 0,
    "Scalar::Util" => 0,
    "Tree::Simple" => 0,
    "Try::Tiny" => 0,
    "base" => 0,
    "namespace::autoclean" => "0.09",
    "re" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Data::Dumper" => 0,
    "File::Basename" => 0,
    "File::Find" => 0,
    "File::Temp" => 0,
    "HTML::Tidy" => 0,
    "IO::File" => 0,
    "Test::Deep" => 0,
    "Test::Differences" => 0,
    "Test::Fatal" => 0,
    "Test::More" => "0.88",
    "Tree::Simple::Visitor::ToNestedArray" => 0,
    "lib" => 0
  },
  "VERSION" => "0.24",
  "test" => {
    "TESTS" => "t/*.t t/Handler/*.t t/Handler/HTMLStream/*.t t/Simple/*.t t/markup/github/*.t t/markup/multi-dialect/*.t t/markup/standard/*.t t/markup/theory/*.t"
  }
);


my %FallbackPrereqs = (
  "Class::Load" => 0,
  "Data::Dumper" => 0,
  "Digest::SHA1" => 0,
  "Exporter" => 0,
  "File::Basename" => 0,
  "File::Find" => 0,
  "File::Slurp" => 0,
  "File::Temp" => 0,
  "Getopt::Long::Descriptive" => 0,
  "HTML::Stream" => 0,
  "HTML::Tidy" => 0,
  "IO::File" => 0,
  "IO::Handle" => 0,
  "List::AllUtils" => 0,
  "Moose" => "0.92",
  "Moose::Meta::Class" => 0,
  "Moose::Role" => 0,
  "MooseX::Getopt::Dashes" => 0,
  "MooseX::Params::Validate" => 0,
  "MooseX::Role::Parameterized" => 0,
  "MooseX::SemiAffordanceAccessor" => "0.05",
  "MooseX::StrictConstructor" => "0.08",
  "MooseX::Types" => "0.20",
  "MooseX::Types::Combine" => 0,
  "MooseX::Types::Moose" => 0,
  "Scalar::Util" => 0,
  "Test::Deep" => 0,
  "Test::Differences" => 0,
  "Test::Fatal" => 0,
  "Test::More" => "0.88",
  "Tree::Simple" => 0,
  "Tree::Simple::Visitor::ToNestedArray" => 0,
  "Try::Tiny" => 0,
  "base" => 0,
  "lib" => 0,
  "namespace::autoclean" => "0.09",
  "re" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



